docs/tutorials/blog.md

```diff
@@ -203,12 +203,8 @@ type Post = {
   title: string;
 };
 
-type LoaderData = {
-  posts: Array<Post>;
-};
-
 export const loader = async () => {
```

Comment:
I can make that change, though the rest of the tutorial uses the const function definitions in several locations. Should I change all of the examples in the tutorial to make it consistent? It seems a bit beyond the scope of the reason for my PR, I just wanted to make sure others wouldn't be tripped up by a type issue while working through the tutorial. I couldn't find the right solution in the documentation, and didn't figure out the right method to fix the issue until after opening the issue.

https://github.com/remix-run/remix/blob/493b6a06d483562840ffc5e41311f8c7d5bb5c7c/docs/tutorials/blog.md?plain=1#L136

https://github.com/remix-run/remix/blob/493b6a06d483562840ffc5e41311f8c7d5bb5c7c/docs/tutorials/blog.md?plain=1#L296

https://github.com/remix-run/remix/blob/493b6a06d483562840ffc5e41311f8c7d5bb5c7c/docs/tutorials/blog.md?plain=1#L456

https://github.com/remix-run/remix/blob/493b6a06d483562840ffc5e41311f8c7d5bb5c7c/docs/tutorials/blog.md?plain=1#L481

https://github.com/remix-run/remix/blob/493b6a06d483562840ffc5e41311f8c7d5bb5c7c/docs/tutorials/blog.md?plain=1#L519

https://github.com/remix-run/remix/blob/493b6a06d483562840ffc5e41311f8c7d5bb5c7c/docs/tutorials/blog.md?plain=1#L553

https://github.com/remix-run/remix/blob/493b6a06d483562840ffc5e41311f8c7d5bb5c7c/docs/tutorials/blog.md?plain=1#L668

https://github.com/remix-run/remix/blob/493b6a06d483562840ffc5e41311f8c7d5bb5c7c/docs/tutorials/blog.md?plain=1#L750

Reply:
@johnhooks we need to update the docs to use the new inference since the 1.6.5 release, but we didn't find the time to do it. 

I wouldn't mind at all if you extended this PR scope to tackle the whole update. WDYT ?

Reply:
@machour I will definitely take that on. I assume that means most of the function definitions don't need to be typed as `LoaderFunction` but should instead have the args typed as `LoaderArgs`. Is there a better way to make sure the modified code examples pass the type checker, rather than copy and pasting into an actual remix project? Just want to make sure I'm not missing some handy way to type check in Markdown.

Reply:
🙏
Exactly. Read through the 1.6.5 announcement and you'll see the ideal syntax we're looking for, for both loader and action. 
As @jenseng said, typing using `args` instead of `{}` should satisfy the linter. 
I'd really appreciate if you took the time to make the tutorial run locally and update the docs accordingly (you can even share the repo here so that we give it a look)

Reply:
Also how are the line numbers in all the code snippets being generated? Some of the changes, especially adding `import type { LoaderArgs } from "@remix-run/node";` to the top of the example, will mess all of those up. Is there an app example folder that has the code examples in them? I know the Indie Stack is used, but this tutorial adds the `post.tsx` file.

https://github.com/remix-run/remix/blob/493b6a06d483562840ffc5e41311f8c7d5bb5c7c/docs/tutorials/blog.md?plain=1#L132

Reply:
Line numbers gets added automatically, and the `lines=[1-2,4-17]` syntax allow to highlight the interesting lines (here, line 1 and 2, and line 4 to 17).

You'll have to count lines with your fingers to find the good values (and maybe bring up some nice childhood memories 😅)

Reply:
Sorry for all the questions, but I just want to make sure while making all these changes I keep the appropriate intent of the tutorial. Because of the new type inference some sections about making Typescript happy aren't required. Though I could intentionally leave out the generic argument in the first use of `useLoaderData<typeof loader>()`, let it error and use the situation to explain that `useLoaderData` needs a generic argument in order to resolve the type. What do you think?

Reply:
@machour I am going though everything, making sure the examples pass the type checker, and that the tutorial still makes sense with all the example edits. The repo is https://github.com/remix-run/remix, I'm still working on the changes, right now there isn't anything pushed to it to look at, beyond what's already in the PR. I'll let you know when I get it up.

Reply:
Ask as much questions as you need.

Since you were originally confused about this particular part, the approach you're proposing makes a lot of sense imo. 

We will review what you'll come up with and maybe adjust a thing or two, so don't worry and do as you see fit for now.